Source/core/IIterator.h

```diff
@@ -174,7 +174,7 @@ namespace Core {
 
         inline ELEMENT operator->()
         {
-            ASSERT(IsValid());
```

Comment:
What is the use of this change? This is a copy of what is in IsValid(). So now the logic is spread.

Reply:
This (also learned it during the conf -call) is a compilation issue caused in GCC 9.3.0 but only in the RDK build environment (not in any other environment). Information is send, Metrological will also deep dive into this to see what the root casue is. For now it reolves the issue in the RDK build but does require a further investigation. INternal Jira ticket will be raised.